### PR TITLE
automation: add build with carpenter

### PR DESCRIPTION
**Pull request by sandrobonazzola**
## Changes introduced with this PR

automation: add build with carpenter

---
By contributing to this repository, I agree to the [contribution guidelines](https://github.com/arcalot/.github/blob/main/CONTRIBUTING.md).

Signed-off-by: Sandro Bonazzola <sbonazzo@redhat.com>

### CODE REVIEW

**Review by portante — CHANGES_REQUESTED** [review]
On the surface this looks okay, but the tests failed due to missing `git`.  Is this something that should also be failing with the example plugin?  What has changed?

Further, is it sufficient to install the `python39` module to ensure when the `pip3` and `python3` commands are invoked they are actually running Python 3.9?

**Comment by sandrobonazzola** [comment]
> On the surface this looks okay, but the tests failed due to missing `git`. Is this something that should also be failing with the example plugin? What has changed?

I need a newer sdk than the one shipped in pip repository, that's why git is required.
Fixed.

> 
> Further, is it sufficient to install the `python39` module to ensure when the `pip3` and `python3` commands are invoked they are actually running Python 3.9?

yes, adding module 3.9 is enough for having python3 mapped to python 3.9 binary.

**Review by ghost — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
 
 VOLUME /config
 
-ENTRYPOINT ["/usr/local/bin/python3", "/app/wait_plugin.py"]
-CMD ["-f", "/config/wait.yaml"]
\ No newline at end of file
+ENTRYPOINT ["python3", "/app/wait_plugin.py"]
+CMD []
+
+LABEL org.opencontainers.image.source="https://github.com/arcalot/arcaflow-plugin-wait"
```
Comment (portante) [inline Dockerfile:20]: Still says `wait`.

I have lots of reservations about separate repositories still.

File: Dockerfile [hunk 2]
```diff
@@ -1,15 +1,25 @@
...
+LABEL org.opencontainers.image.licenses="Apache-2.0+GPL-2.0-only"
+LABEL org.opencontainers.image.vendor="Arcalot project"
+LABEL org.opencontainers.image.authors="Arcalot contributors"
+LABEL org.opencontainers.image.title="Python Plugin Template"
```
Comment (portante) [inline Dockerfile]: `s/Template/Wait`?

**Review by portante — CHANGES_REQUESTED** [review]
Why are we not first updating the SDK in PyPI instead of adding a `git` dependency here?

**Comment by sandrobonazzola** [comment]
> Why are we not first updating the SDK in PyPI instead of adding a `git` dependency here?

@janosdebugs wants to add some more stuff there before releasing.

File: Dockerfile (hunk 2 above)
Comment (sandrobonazzola) [reply Dockerfile]: right, fixing

Comment (sandrobonazzola) [reply Dockerfile]: done

**Comment by portante** [comment]
> @janosdebugs wants to add some more stuff there before releasing.

How long will that take?

**Review by portante — DISMISSED** [review]
Waiting to hear the timeline for the PyPI updates vs adding another dependency on `git` that will rot on the vine in the future ...

**Review by mfleader — APPROVED** [review]
I like it!

**Comment by ghost** [comment]
> Waiting to hear the timeline for the PyPI updates vs adding another dependency on `git` that will rot on the vine in the future ...

It won't, but we are still missing the docs for the new release.

**Pull request merged**